xarray/core/rolling.py

```diff
@@ -973,7 +973,9 @@ def construct(
             else:
                 reshaped[key] = var
 
-        should_be_coords = set(window_dim) & set(self.obj.coords)
+        should_be_coords = (set(window_dim) & set(self.obj.coords)) | set(
+            self.obj.coords
+        )
```

Comment:
```suggestion
        should_be_coords = set(window_dim) | set(self.obj.coords)
```

Reply:
I tried that but it doesn't work - it fails if window_dim is a dimension but not a dimension coordinate.

Reply:
thanks!